### DISCUSSION TITLE
unable to deploy code server on railway

### DISCUSSION THREAD

**Discussion by leamandeep** [coder/code-server#4083] [Q&A]
i tried a lot still unable to deploy code server on railway . it still showing 404 error. plz help me as early as possible .
![Screenshot_2021-09-01_135248](https://user-images.githubusercontent.com/68345524/131638468-7cefc92c-d7b4-4223-9463-f7d19040a60a.jpg)

**Comment by lg-nineoneninethreenine** [comment]
Hey @amandeep2603 try using the consistent URL you get in Railway, or wait for some time and try again. This occurs sometimes in Railway but will be fixed. Till then, wait, and then use the first URL Railway gives, ie. https://<[your-railway-project-name]>-production-<[strange-code]>.up.railway.app

**Reply by lg-nineoneninethreenine** [reply]
That was for me. It might be different for you. Can you share a Railway screenshot?

**Reply by jsjoeio** [reply]
Heads up @amandeep2603 I deleted your picture because it contained your password. You don't want to expose that.

Did you follow @im-coder-lg's advice?

It might be worth sharing a video so we can see exactly what's happening (don't show your password though)

**Reply by lg-nineoneninethreenine** [reply]
The error is a 404 and I saw the logs only. Some strange error was there.

**Reply by leamandeep** [reply]
yeah!! you are right i also felt strange.

**Reply by leamandeep** [reply]
should i share video??

**Comment by leamandeep** [comment]
i am sharing my screen recorded video . Also i have  hided my password.

https://user-images.githubusercontent.com/68345524/131779852-7a250613-7489-4a36-9205-c6767a53a7c6.mp4

**Reply by lg-nineoneninethreenine** [reply]
yo @amandeep2603 see that:
![image](https://user-images.githubusercontent.com/76845820/131780173-356807fc-6dd7-4c83-b266-98a2107ce717.png)
I have NEVER spotted that error(something 200 so and so) for my entire life. I use a Railway deployment. This is not supposed to happen. @jsjoeio has this error started pestering everyone in deploy-code-server? Because he is using that repo as the template. Looks like I am gonna get into test mode. @amandeep2603 wait and let me see if I could reproduce this.

**Comment by lg-nineoneninethreenine** [comment]
Ladies and gentlemen, this is sus because I used the LATEST COMMIT of deploy-code-server and am editing the photo as we speak.
Either @amandeep2603 has some strange config error, or it has been fixed and he has to use another method to get it up and running.
@jsjoeio can you recreate this and make a video on that? I can't record as of now. I will share my screenshot, and you share the recreate video.

**Reply by lg-nineoneninethreenine** [reply]
[![oh yeah 2 0](https://user-images.githubusercontent.com/76845820/131781663-a08418f4-59c4-4b98-8037-b04544e12795.PNG)](https://code-server-test-001-production.up.railway.app/)
hit that and you will be redirected. Use the password I gave, it is a test so I am deleting it after a week.
Password: anything_as_a_password

**Reply by lg-nineoneninethreenine** [reply]
and anything_as_a_password is literally the password

**Comment by lg-nineoneninethreenine** [comment]
/ping @jsjoeio and @amandeep2603 
I did it.
![image](https://user-images.githubusercontent.com/76845820/131797783-10295b33-df5a-48a6-81b5-9e4b80f0c1ed.png)
See that :arrow_up: 
I forked deploy-code-server, changed the Railway button data and voila! Recreated the ORIGINAL!

**Reply by lg-nineoneninethreenine** [reply]
Lookie here: https://code-server-test-002-production.up.railway.app
Successfully recreated. Looks like @amandeep2603 forked/did something and ended up with the error.
If you see the Railway docs in my fork of deploy-code-server, you can see that I modified the Railway button. Then, after hitting and deploying, you end up with the issue. That is the problem, I guess.

**Reply by lg-nineoneninethreenine** [reply]
[`This one`](https://github.com/cdr/code-server/discussions/4083#discussioncomment-1269683) was from the official deployment repo.
But [`this one`](https://github.com/cdr/code-server/discussions/4083#discussioncomment-1270082) was from the fork.

**Reply by lg-nineoneninethreenine** [reply]
I realised something. Look at this:
![image](https://user-images.githubusercontent.com/76845820/131798949-43a5ec00-cfb9-4384-838c-eb5ec4345093.png)
Something is happening. Look at this image and then go to the URL. How is code-server even working?

**Reply by leamandeep** [reply]
> /ping @jsjoeio and @amandeep2603
> I did it.
> ![image](https://user-images.githubusercontent.com/76845820/131797783-10295b33-df5a-48a6-81b5-9e4b80f0c1ed.png)
> See that ⬆️
> I forked deploy-code-server, changed the Railway button data and voila! Recreated the ORIGINAL!

is it working or not ??

**Reply by lg-nineoneninethreenine** [reply]
Confused, really. At first, it did work but then it might have patched the error up. I got a fork of deploy-code-server, go to the Railway section in that and hit the Railway button. Let's try that. If it fails, it is problems on your account or something happened that time.

**Reply by jsjoeio** [reply]
> I got a fork of deploy-code-server, go to the Railway section in that and hit the Railway button. Let's try that. If it fails, it is problems on your account or something happened that time.

@amandeep2603 give that a try and report back to us! I'm not sure what could be happening 🤔
